language-plutus-core/plutus-ir/Language/PlutusIR/Value.hs

```diff
@@ -4,11 +4,15 @@ module Language.PlutusIR.Value (isTermValue) where
 import           Language.PlutusIR
 
 -- | Whether the given PIR term is (will compile to) a PLC term value. Very similar to
--- the PLC definition.
+-- the PLC definition. Crucially, evaluating a value should return the value immediately,
+-- and it should not be possible for it to loop or throw error.
 isTermValue :: Term tyname name uni a -> Bool
 isTermValue = \case
     LamAbs {} -> True
     TyAbs {} -> True
     Constant {} -> True
+    -- Variables are values: they can only refer to entries in the environment, which
+    -- are always values
+    Var {} -> True
```

Comment:
WAT. Pinging @jmchapman

Reply:
Oh well, in "value restriction" a variable is a value [indeed](http://mlton.org/ValueRestriction).

Reply:
Am I crazy here? Maybe I don't mean "value", but I remain confused about what the true meaning of "value" is. Perhaps I need a different notion: "will not compute" or something. A variable has that property since, while it may have something else substituted in for it, that will itself be a value.

Reply:
I think you're perfectly correct here and I just needed to educate myself on what "value" means in "value restriction". I thought it was "a term in WHNF", but apparently not.

> A variable has that property since, while it may have something else substituted in for it, that will itself be a value.

I'd call an act of looking up a variable in the environment a step of computation.

Reply:
I think I should anyway change this to talk about something more clearly defined. Since in this case we actually care about side-effect-freeness.